openedx/core/djangoapps/course_roles/helpers.py

```diff
@@ -0,0 +1,155 @@
+"""
+Helpers for the course roles djangoapp which is used for authorization.
+"""
+from django.contrib.auth.models import AnonymousUser, User  # lint-amnesty, pylint: disable=imported-auth-user
+from django.utils.translation import gettext as _
+
+from edx_toggles.toggles import WaffleFlag
+from openedx.core.djangoapps.course_roles.models import UserRole
+from xmodule.modulestore.django import modulestore
+
+
+# .. toggle_name: FLAG_USE_PERMISSION_CHECKS
+# .. toggle_implementation: WaffleFlag
+# .. toggle_default: False
+# .. toggle_description: Enabling the toggle will allow the db checks for a users permissions. These are used alongside current
+#   roles checks. If the flag is not enabled, only the roles checks will be used.
+# .. toggle_use_cases: temporary
+# .. toggle_creation_date: 2023-10-17
+# .. toggle_target_removal_date: 2023-12-01
+# .. toggle_warning:
+USE_PERMISSION_CHECKS_FLAG = WaffleFlag('course_roles.use_permission_checks', module_name=__name__) or False  # lint-amnesty, pylint: disable=toggle-missing-annotation
+
+
+def use_permission_checks():
+    """
+    Returns ture if permissions checks should be used
+    """
+    return USE_PERMISSION_CHECKS_FLAG.is_enabled()
+
+
+def user_has_permission_course(user, permission_name, course_id):
```

Comment:
It's a little awkward how people have to pass in the `.value` at the end, e.g. `CourseRolesPermission.MODERATE_DISCUSSION_FORUMS.value`. I think most would think to pass in `CourseRolesPermission.MODERATE_DISCUSSION_FORUMS` instead.